### PR TITLE
fix(ui-library): fix icon button events

### PR DESCRIPTION
**Pull request by davidken91**

### CODE REVIEW

File: packages/js-example-app/src/index.js [hunk 1]
```diff
@@ -5,7 +5,8 @@ const toggleLoadingButton = document.querySelector('#toggleLoadingState');
 const toggleDisabledState = document.querySelector('#toggleDisabledState');
 const logsContainer = document.querySelector('#logs');
 
-const blrButton = document.getElementsByTagName('blr-button-text')[0];
+const blrTextButton = document.getElementsByTagName('blr-text-button')[0];
+const blrIconButton = document.getElementsByTagName('blr-buton-icon')[0];
```
Comment (angsherpa456) [inline packages/js-example-app/src/index.js:9]: name inconsistency in the tag ('blr-buton-icon') and the const variable, please check it once?

File: packages/ui-library/src/components/button-icon/index.stories.ts [hunk 2]
```diff
@@ -121,23 +121,26 @@ export default {
       },
     },
     // Events
-    onChange: {
-      description: 'Fires when the value changes.',
-      action: 'onChange',
+    blrClick: {
+      name: 'blrClick',
```
Comment (angsherpa456) [inline packages/ui-library/src/components/button-icon/index.stories.ts:125]: name is not required, can be removed. Same for the following events as well.

File: packages/js-example-app/src/index.js (hunk 1 above)
Comment (davidken91) [reply packages/js-example-app/src/index.js]: Well spotted. That's fixed now

**Review by bar-tay — DISMISSED** [review]
lgtm :)

**Review by JpunktWpunkt — DISMISSED** [review]

File: packages/js-example-app/src/index.ejs [hunk 3]
```diff
@@ -34,16 +34,16 @@
           <blr-button-icon
             theme="Light"
             variant="primary"
-            sizevariant="md"
+            size-variant="md"
             icon="blr360"
-            arialabel="Icon Button"
-            iconbuttonid="iconButtonId">
+            arialabel="Button Icon"
```
Comment (angsherpa456) [inline packages/js-example-app/src/index.ejs:39]: should be aria-label

File: packages/js-example-app/src/index.ejs [hunk 4]
```diff
@@ -34,16 +34,16 @@
...
+            button-icon-id="buttonIconId">
           </blr-button-icon>
         </div>
 
         <div class="component">
           <p>Select</p>
-          <blr-select label="Boiler Select Component">
+          <blr-select theme="Light" size-variant="md" has-label="true" label="Label-text" label-appendix="(Appendix)" icon="blrChevronDown" arialabel="Select" select-id="selectId" name="select">
```
Comment (angsherpa456) [inline packages/js-example-app/src/index.ejs:46]: aria-label here as well.

File: packages/js-example-app/src/index.ejs [hunk 5]
```diff
@@ -56,49 +56,81 @@
 
         <div class="component">
           <p>Checkbox</p>
-          <blr-checkbox has-label="true" label="Please check" check-input-id="Test123"></blr-checkbox>
+          <blr-checkbox
+            theme="Light"
+            size="md"
+            has-label="true"
+            label="Label-text"
+            arialabel="check Input"
```
Comment (angsherpa456) [inline packages/js-example-app/src/index.ejs:64]: aria-label here as well

File: packages/js-example-app/src/index.ejs [hunk 6]
```diff
@@ -56,49 +56,81 @@
...
             label-appendix="(Appendix)"
-            input-field-number-id="test-id"
-            theme="Light"
-            name="InputFieldNumber"
-            decimals="10"
-            leading-zeros="3"
-          ></blr-input-field-number>
+            has-icon="true"
+            icon="blr360"
+            arialabel="InputFieldText"
```
Comment (angsherpa456) [inline packages/js-example-app/src/index.ejs:84]: aria-label here as well

File: packages/js-example-app/src/index.ejs [hunk 7]
```diff
@@ -56,49 +56,81 @@
...
+            min-length="10"
             max-length="140"
+            has-label="true"
+            label="Label-text"
+            label-appendix="(Appendix)"
+            has-counter="true"
             warning-limit-type="warningLimitInt"
             warning-limit-int="105"
-          ></blr-textarea>
+            arialabel="Text Area"
```
Comment (angsherpa456) [inline packages/js-example-app/src/index.ejs:130]: aria-label here as well

File: packages/js-example-app/src/index.ejs (hunk 3 above)
Comment (davidken91) [reply packages/js-example-app/src/index.ejs]: Ok. Fixed

File: packages/js-example-app/src/index.js [hunk 8]
```diff
@@ -31,27 +32,39 @@ toggleLoadingButton.addEventListener('click', () => {
...
+});
+
+blrButtonIcon.addEventListener('blrClick', () => {
+  addLog('blr-icon-button clicked');
 });
 
-blrButton.addEventListener('blrFocus', () => {
-  addLog('blr-button-text focused');
+blrButtonIcon.addEventListener('blrFocus', () => {
+  addLog('blr-icon-button focused');
```
Comment (angsherpa456) [inline packages/js-example-app/src/index.js:63]: should be blr-button-icon for consistency.

File: packages/js-example-app/src/index.js [hunk 9]
```diff
@@ -31,27 +32,39 @@ toggleLoadingButton.addEventListener('click', () => {
...
 });
 
-blrButton.addEventListener('blrBlur', () => {
-  addLog('blr-button-text blurred');
+blrButtonIcon.addEventListener('blrBlur', () => {
+  addLog('blr-icon-button blurred');
```
Comment (angsherpa456) [inline packages/js-example-app/src/index.js:67]: should be blr-button-icon for consistency.

File: packages/js-example-app/src/index.js [hunk 10]
```diff
@@ -31,27 +32,39 @@ toggleLoadingButton.addEventListener('click', () => {
...
+blrButtonText.addEventListener('blrFocus', () => {
+  addLog('blr-text-button focused');
+});
+
+blrButtonText.addEventListener('blrBlur', () => {
+  addLog('blr-text-button blurred');
+});
+
+blrButtonIcon.addEventListener('blrClick', () => {
+  addLog('blr-icon-button clicked');
```
Comment (angsherpa456) [inline packages/js-example-app/src/index.js:59]: should be blr-button-icon for consistency.

File: packages/js-example-app/src/index.js [hunk 11]
```diff
@@ -31,27 +32,39 @@ toggleLoadingButton.addEventListener('click', () => {
...
 });
 
-blrButton.addEventListener('blrClick', () => {
-  addLog('blr-button-text clicked');
+blrButtonText.addEventListener('blrClick', () => {
+  addLog('blr-text-button clicked');
+});
+
+blrButtonText.addEventListener('blrFocus', () => {
+  addLog('blr-text-button focused');
```
Comment (angsherpa456) [inline packages/js-example-app/src/index.js:51]: should be blr-button-text for consistency.

File: packages/js-example-app/src/index.js [hunk 12]
```diff
@@ -31,27 +32,39 @@ toggleLoadingButton.addEventListener('click', () => {
...
+});
+
+blrButtonText.addEventListener('blrBlur', () => {
+  addLog('blr-text-button blurred');
```
Comment (angsherpa456) [inline packages/js-example-app/src/index.js:55]: should be blr-button-text for consistency.

File: packages/js-example-app/src/index.js (hunk 8 above)
Comment (davidken91) [reply packages/js-example-app/src/index.js]: Ok, fair enough. That has been resolved

**Review by angsherpa456 — APPROVED** [review]
LGTM

**Review by angsherpa456 — APPROVED** [review]
LGTM

**Review by thrbnhrtmnn — APPROVED** [review]
Looks good in Storybook and works for me

**Pull request merged**